backend/src/spectral_equation.py

```diff
@@ -0,0 +1,116 @@
+"""spectral_equation.py
+Collection of functions related to optimizing/solving the spectral matrix
+equation for a given run
+
+Functions:
+    xyztolab  : convert a list of xyz values to lab values
+    ciede2000 : get ciede2000 color differences for two lists of lab values
+
+Authors:
+    Brendan Grau <https://github.com/Victoriam7>
+
+License:
+    © 2022 BeyondRGB
+    This code is licensed under the MIT license (see LICENSE.txt for details)
+"""
+import math
+import numpy as np
+
+
+def _f(x):
+    # TODO verify this is good code
+    if x > 216.0/24389.0:
+        return x**(1.0/3.0)
+    else:
+        return ((24389.0/27.0)*x+16.0)/116.0
+
+
+def xyztolab(xyzs):
+    # TODO verify this is good code
+    labs = np.ndarray(xyzs.shape)
+    for i in range(0, xyzs.shape[1]):
+        xyz = xyzs[:, i]
+        labs[0, i] = 116*_f(xyz[1]/100)-16
+        labs[1, i] = 500*(_f(xyz[0]/96.7206275)-_f(xyz[1]/100))
+        labs[2, i] = 200*(_f(xyz[1]/100)-_f(xyz[2]/81.4280151))
+    return labs
+
+
+def ciede2000(Lab_1, Lab_2):
+    """ get ciede2000 color differences for two lists of lab values
+
+    There is a lot of work to do here. Nothing should be saved. This is garbage
+    copy paste code that might not even be correct. Ether way, this needs to be
+    implemented in a sane way and on numpy arrays instead of single values. When
+    this is fixed do not forget to update the xyz equation in calibration.py to
+    use lists.
+
+    INUPTS AND OUTPUTS GO HERE
+    """
+    C_25_7 = 6103515625 # 25**7
```

Comment:
If we want people to keep working on this, shouldn't we try to make variable names a little clearer? I understand this part will be rewritten, but I honestly don't know where I'd start with these variable names and lack of comments